### ISSUE TITLE
Endless 'Loading Moonlight plugin'

### ISSUE THREAD

**Issue by Gamer9** [moonlight-stream/moonlight-chrome#102]
Hello, I recently tried Moonlight Chrome v0.5 (Alpha 5) on W7 with Chrome v53.0.2785.89 m both (64-bit) and (32-bit), and have encountered an error.

On starting 'Loading Moonlight plugin' continues without end and this display persists through the other options/screens below:
Clicking on Back <-- Moonlight still shows 'Loading...' on main Moonlight screen.
Clicking 'ADD HOST' still shows 'Loading...'  in background of 'Add PC Manually' pop up.
Entering 'IP Address or Hostname of Geforce PC' (or even leaving blank) and clicking Continue brings up the following 2 messages:

ERROR: cert has not been generated yet. Is NaCl initialized?
pairing to failed!

Previously Moonlight Chrome v0.4 ran fine on Chrome from 2015, (sorry don't know version as I updated to the latest version at same time as Moonlight Chrome v0.5). Moonlight Chrome v0.4 still loads fine on latest Chrome (32-bit) so I don't think it's the change of Chrome version that's the issue.

Any ideas what could be causing this and how to resolve?

If you require any further specific tests or info please advise.

Thanks in advance,
Islam

**Comment by raidancampbell** [comment]
can you remove the app, re-add it, and paste the output of the javascript console?

**Comment by Gamer9** [comment]
Hello, I tried as you asked but the Console didn't give any info. However I feel I must apologize as I have identified the issue. It seems my network was disconnected at the time which was causing the endless loading. I feel I should close/delete this post but I'd like to take the time to provide you some additional info on this issue and it's solution that may or may not be useful to you.

As I mentioned earlier, on Chrome 32bit (under W7 32bit), v0.4 did work fine in the same condition (ie of a disconnected/LAN only network) whereas v.05 did not.

Why it worked on one but not the other was very confusing to me.

So on Chrome 64bit (under W7 64bit) I installed v0.4 just to see if it differs, but like v0.5 it too produced the endless loading until an active connection was made.

Also, the solution between the two Moonlight versions seems slightly different.

v0.4 successfully loaded I think with only a LAN connection whereas I think v0.5 needed an internet connection. However, I couldn't retest this to confirm because of the below.

It seems that after the first successful loading, the status of the network connection after that no longer matters. ie whether there's No Connection/LAN Only/Internet, after the initial successful loading, Moonlight Chrome will always load successfully. It didn't matter if I removed/reinstalled Moonlight or even uninstalled/reinstalled Chrome, Moonlight always continued to load successfully. (hence why I was unable to retest if LAN / Internet connection was the solution but i think it's LAN for v0.4 and Internet (or maybe a very long LAN wait i.e a minute or two) for v0.5)
 
It seems to me that at a minimum Moonlight should load successfully on the first attempt when a LAN connection is active (or perhaps warn/hint if LAN/Internet is necessary for the intended connection type).

If this is all completely irrelevant and is indeed the intended way Moonlight should be working, then again, my apologize for wasting your time. If you think there is a minor issue then feel free to let me know it there's anything I can do to help test.

And thanks for the brilliant software.
